examples/trace/trace_attachments.rb

```diff
@@ -0,0 +1,178 @@
+      role: "user",
+      content: [
+        {
+          type: "text",
+          text: "Example with attachment"
+        },
+        attachment.to_h
+      ]
+    }
+  ]
+
+  span.set_attribute("braintrust.input_json", JSON.generate(messages))
+end
+
+# Initialize Braintrust
+Braintrust.init(blocking_login: true)
+
+# Get a tracer
+tracer = OpenTelemetry.tracer_provider.tracer("attachments-example")
+
+# Create a parent span to wrap all examples
+root_span = nil
+tracer.in_span("examples/trace/trace_attachments.rb") do |span|
+  root_span = span
+
+  # Example 1: Create attachment from bytes
+  puts "Example 1: from_bytes"
+  tracer.in_span("attachment.from_bytes") do |example_span|
+    image_bytes = create_test_image_bytes
+    att = Braintrust::Trace::Attachment.from_bytes(
+      Braintrust::Trace::Attachment::IMAGE_PNG,
+      image_bytes
+    )
+
+    puts "  Created attachment from #{image_bytes.bytesize} bytes"
+    log_attachment(example_span, att)
+  end
+
+  # Example 2: Create attachment from file
+  puts "\nExample 2: from_file"
+  tracer.in_span("attachment.from_file") do |example_span|
+    # Create a temporary file
+    temp_file = Tempfile.new(["test-image", ".png"])
+    begin
+      temp_file.binmode
+      temp_file.write(create_test_image_bytes)
+      temp_file.flush
+      temp_file.close
+
+      att = Braintrust::Trace::Attachment.from_file(
+        Braintrust::Trace::Attachment::IMAGE_PNG,
+        temp_file.path
+      )
+
+      puts "  Created attachment from file"
+      log_attachment(example_span, att)
+    ensure
+      temp_file.unlink
+    end
+  end
+
+  # Example 3: Create attachment from URL
+  puts "\nExample 3: from_url"
+  tracer.in_span("attachment.from_url") do |example_span|
+    # Fetch image from Braintrust's GitHub avatar
+    url = "https://avatars.githubusercontent.com/u/109710255?s=200&v=4"
+
+    begin
+      att = Braintrust::Trace::Attachment.from_url(url)
+      puts "  Fetched attachment from URL"
+      log_attachment(example_span, att)
+    rescue => e
+      puts "  Failed to fetch URL: #{e.message}"
+    end
+  end
+
+  # Example 4: Use attachment in a vision task (manual span logging)
+  puts "\nExample 4: Manual span with attachment (vision task)"
+  tracer.in_span("vision.analyze_image") do |vision_span|
+    # Create attachment from test data
+    image_bytes = create_test_image_bytes
+    att = Braintrust::Trace::Attachment.from_bytes(
+      Braintrust::Trace::Attachment::IMAGE_PNG,
+      image_bytes
+    )
+
+    # Construct a message with text and image (similar to OpenAI/Anthropic format)
+    messages = [
+      {
+        role: "user",
+        content: [
+          {
+            type: "text",
+            text: "What's in this image? Describe it in detail."
+          },
+          att.to_message  # Attachment in correct format
+        ]
+      }
+    ]
+
+    # Log the messages as input
+    vision_span.set_attribute("braintrust.input_json", JSON.generate(messages))
```

Comment:
attachments are implemented on the `gen_ai.input.messages` attribute. we can add it on the braintrust attributes too but right now this won't be parsed